src/mame/drivers/att4425.cpp

```diff
@@ -0,0 +1,300 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+/***************************************************************************
+
+    AT&T model 4425 text terminal with mosaic graphics character set
+    Also known as Teletype model 56D.
+
+    Skeleton driver.  This terminal is supported by netpbm -- pbmto4425(1)
+
+    To do:
+    - keyboard
+    - everything else
+
+****************************************************************************/
+
+
+#include "emu.h"
+
+#include "cpu/z80/z80.h"
+#include "machine/clock.h"
+#include "machine/i8251.h"
+#include "machine/ram.h"
+#include "machine/z80ctc.h"
+#include "machine/z80sio.h"
+
+#include "bus/rs232/rs232.h"
+#include "machine/keyboard.h"
+
+#include "screen.h"
+
+
+#define SCREEN_TAG          "screen"
+#define Z80_TAG             "maincpu"
+#define Z80CTC_TAG          "z80ctc"
+#define Z80SIO_TAG          "z80sio"
+#define RS232_A_TAG         "sioa"
+#define RS232_B_TAG         "siob"
+#define I8251_TAG           "i8251"
+
+class att4425_state : public driver_device
+{
+public:
+	att4425_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, Z80_TAG)
+		, m_i8251(*this, I8251_TAG)
+		, m_sio(*this, Z80SIO_TAG)
+		, m_p_videoram(*this, "videoram")
+		, m_p_chargen(*this, "chargen")
+		, m_screen(*this, "screen")
+		{ }
+
+	DECLARE_PALETTE_INIT(att4425);
+
+	DECLARE_WRITE_LINE_MEMBER(write_line_clock);
+	DECLARE_WRITE_LINE_MEMBER(write_keyboard_clock);
+
+	virtual void machine_start() override;
```

Comment:
Can you please make overrides of protected member function protected, and make data members protected or private?  We're trying to encourage good habits.